ttnn/cpp/ttnn/op_library/binary/binary_op.hpp

```diff
@@ -124,15 +124,20 @@ struct ExecuteBinary {
             input_tensor_b = tt::tt_metal::repeat(input_tensor_b, repeats.value(), output_memory_config);
         }
 
-        return operation::run(
+        DataType dtype = output_dtype.value_or(input_tensor_a.get_dtype());
```

Comment:
Need to find a way to retrieve dtype in a more abstract way to not mention specific op type here before merging.